### PR TITLE
Advertise functorch in vmap docs more

### PR DESCRIPTION
**Pull request by zou3519**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #71945

This PR:
- advertises functorch more in the vmap docs
- advertises functorch more in the vmap warning message

Test Plan:
- check the docs preview

Differential Revision: [D33828950](https://our.internmc.facebook.com/intern/diff/D33828950)

### CODE REVIEW

**Comment by zou3519** [comment]
@zou3519 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33828950).

File: torch/_vmap_internals.py [hunk 1]
```diff
@@ -248,11 +249,12 @@ def vmap(func: Callable, in_dims: in_dims_t = 0, out_dims: out_dims_t = 0) -> Ca
         sequences out of the box.
     """
     warnings.warn(
-        'Please use functorch.vmap instead of torch.vmap '
-        '(https://github.com/pytorch/functorch). '
-        'We\'ve moved development on torch.vmap over to functorch; '
-        'functorch\'s vmap has a multitude of significant performance and '
-        'functionality improvements.',
+        'torch.vmap is no longer being maintained. Please use '
```
Comment (Chillee) [inline torch/_vmap_internals.py:252]: I would remove this terminology :P I think the old terminology of "we've moved development over to functorch" is better.

**Review by Chillee — APPROVED** [review]
LGTM

File: torch/_vmap_internals.py (hunk 1 above)
Comment (zou3519) [reply torch/_vmap_internals.py:252]: Hmm, if you think the old terminology is better then I probably won't merge this PR then

**Pull request closed without merging**